cdap-app-fabric/src/main/java/co/cask/cdap/templates/AdapterSpecification.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates;
+
+import co.cask.cdap.api.data.stream.StreamSpecification;
+import co.cask.cdap.api.schedule.ScheduleSpecification;
+import co.cask.cdap.api.worker.WorkerSpecification;
+import co.cask.cdap.data.dataset.DatasetCreationSpec;
+import com.clearspring.analytics.util.Preconditions;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.JsonElement;
+import com.sun.istack.Nullable;
+
+import java.util.Map;
+
+/**
+ * Specification of an adapter.
+ */
+public final class AdapterSpecification {
+  private final String name;
+  private final String description;
+  private final String template;
+  private final ScheduleSpecification scheduleSpec;
+  private final WorkerSpecification workerSpec;
+  private final Map<String, StreamSpecification> streams;
```

Comment:
Same question as the before. Do we need to this info so that platform will create the streams, datasets? Or will the source/sink be able to do it themselves?